MoreLinq/EndsWith.cs

```diff
@@ -29,21 +29,24 @@ static partial class MoreEnumerable
         /// <typeparam name="T">Type of elements.</typeparam>
         /// <param name="first">The sequence to check.</param>
         /// <param name="second">The sequence to compare to.</param>
+        /// <param name="comparer">Optional equality comparer, defaults to <see cref="EqualityComparer{T}.Default" />.</param>
         /// <returns>
         ///     <c>true</c> if <paramref name="first" /> ends with elements equivalent to <paramref name="second" />.
         /// </returns>
         /// <remarks>
         ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.EndsWith(string)" /> and
-        ///     it calls <see cref="object.Equals(object)" /> on pairs of elements at the same index.
+        ///     it calls <see cref="IEqualityComparer{T}.Equals(T,T)" /> on pairs of elements at the same index.
         /// </remarks>
-        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
+        public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second, IEqualityComparer<T> comparer = null)
         {
             if (first == null) throw new ArgumentNullException("first");
             if (second == null) throw new ArgumentNullException("second");
 
+            comparer = comparer ?? EqualityComparer<T>.Default;
+
             using (var firstIter = first.Reverse().GetEnumerator())
```

Comment:
@fsateler You would probably have to commit the second sequence to memory in that case and one would imagine that it's usually smaller. However, you end up paying that cost up front if, for example, the first sequence is smaller in size. In any event, it would be better than the current implementation.

@anjdreas Sure but then you have to go the distance. :) What should be avoided is duplicate code. If it's about mere convenience using a naïve implementation then such a one-liner convenience extension method that builds on simply on top of other methods should be added to the user code.